### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by erikagayan**

### CODE REVIEW

**Review by YuriiKindrat — CHANGES_REQUESTED** [review]
image
fix that

**Comment by erikagayan** [comment]
brute_force_password

File: app/main.py [hunk 1]
```diff
@@ -20,8 +21,33 @@ def sha256_hash_str(to_hash: str) -> str:
...
+    for num in range(start, end):
+        password = str(num).zfill(8)
+        if check_password(password):
+            return password
+    return None
+
+
+def brute_force_password():
+    num_processes = cpu_count()  # Number of available CPU cores
+    chunk_size = 10000000  # Number of combinations each process will check
```
Comment (LiudmylaKulinchenko) [inline app/main.py:39]: Remove comments

File: app/main.py [hunk 2]
```diff
@@ -20,8 +21,33 @@ def sha256_hash_str(to_hash: str) -> str:
...
 
 
-def brute_force_password() -> None:
-    pass
+def check_password(password):
+    if sha256_hash_str(password) in PASSWORDS_TO_BRUTE_FORCE:
+        return password
+
+
+def brute_force_passwords(start, end):
```
Comment (LiudmylaKulinchenko) [inline app/main.py:29]: Add type annotations everywhere

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

**Review by YuriiKindrat — CHANGES_REQUESTED** [review]
it seems to me, that u have only 6 passwords in your screenshot
there should be 10

**Review by YuriiKindrat — APPROVED** [review]

**Pull request closed without merging**